From 362235f0cf05d758e0c1652e6725700798d65510 Mon Sep 17 00:00:00 2001
From: Igor Grinberg <igrinberg@habana.ai>
Date: Tue, 03 Aug 2021 15:56:41 +0300
Subject: [PATCH] [SW-0] gaudi: security: fix mmu_up and stlb

Commits:
3ead641 ("[SW-47071] gaudi: sec: protecting mmu_up block regs")
885c713 ("[SW-47071] gaudi: sec: protecting stlb block regs")

Have attempted to restrict access to MMU_UP and STLB blocks using
privilege protection bits, but mistakenly used bitwise OR instead of
bitwise AND. This led to keeping all registers in the protection area
unprotected.
Fix this problem by changing the bitwise OR with AND.

Change-Id: I6688f0014b4abf1c040e25a0d6d0f414090123a8
Signed-off-by: Igor Grinberg <igrinberg@habana.ai>
---

diff --git a/arch/gaudi/arm/security.c b/arch/gaudi/arm/security.c
index 8740430..c5aa6d1 100644
--- a/arch/gaudi/arm/security.c
+++ b/arch/gaudi/arm/security.c
@@ -886,15 +886,15 @@
 
 	/* blocking regs within offsets: 0 - 0x80 */
 	prot_base = base;
-	mask = (PROT_BIT(mmMMU_UP_FEATURE_ENABLE) |
-		PROT_BIT(mmMMU_UP_VA_ORDERING_MASK_31_7) |
-		PROT_BIT(mmMMU_UP_VA_ORDERING_MASK_31_7) |
-		PROT_BIT(mmMMU_UP_LOG2_DDR_SIZE) |
-		PROT_BIT(mmMMU_UP_SCRAMBLER) |
-		PROT_BIT(mmMMU_UP_MEM_INIT_BUSY) |
-		PROT_BIT(mmMMU_UP_DBG_MEM_WRAP_RM) |
-		PROT_BIT(mmMMU_UP_PIPE_CREDIT) |
-		PROT_BIT(mmMMU_UP_SLICE_CREDIT) |
+	mask = (PROT_BIT(mmMMU_UP_FEATURE_ENABLE) &
+		PROT_BIT(mmMMU_UP_VA_ORDERING_MASK_31_7) &
+		PROT_BIT(mmMMU_UP_VA_ORDERING_MASK_49_32) &
+		PROT_BIT(mmMMU_UP_LOG2_DDR_SIZE) &
+		PROT_BIT(mmMMU_UP_SCRAMBLER) &
+		PROT_BIT(mmMMU_UP_MEM_INIT_BUSY) &
+		PROT_BIT(mmMMU_UP_DBG_MEM_WRAP_RM) &
+		PROT_BIT(mmMMU_UP_PIPE_CREDIT) &
+		PROT_BIT(mmMMU_UP_SLICE_CREDIT) &
 		PROT_BIT(mmMMU_UP_MMU_BYPASS));
 	regmap_writel(mask, PRIV_PROT_REG(prot_base));
 
@@ -908,15 +908,15 @@
 
 	/* blocking regs within offsets: 0 - 0x80 */
 	prot_base = base;
-	mask = (PROT_BIT(mmSTLB_STLB_FEATURE_EN) |
-		PROT_BIT(mmSTLB_STLB_AXI_CACHE) |
-		PROT_BIT(mmSTLB_HOP_CONFIGURATION) |
-		PROT_BIT(mmSTLB_LINK_LIST) |
-		PROT_BIT(mmSTLB_MEM_CACHE_CONFIG) |
-		PROT_BIT(mmSTLB_SET_THRESHOLD_HOP0) |
-		PROT_BIT(mmSTLB_SET_THRESHOLD_HOP1) |
-		PROT_BIT(mmSTLB_SET_THRESHOLD_HOP2) |
-		PROT_BIT(mmSTLB_SET_THRESHOLD_HOP3) |
+	mask = (PROT_BIT(mmSTLB_STLB_FEATURE_EN) &
+		PROT_BIT(mmSTLB_STLB_AXI_CACHE) &
+		PROT_BIT(mmSTLB_HOP_CONFIGURATION) &
+		PROT_BIT(mmSTLB_LINK_LIST) &
+		PROT_BIT(mmSTLB_MEM_CACHE_CONFIG) &
+		PROT_BIT(mmSTLB_SET_THRESHOLD_HOP0) &
+		PROT_BIT(mmSTLB_SET_THRESHOLD_HOP1) &
+		PROT_BIT(mmSTLB_SET_THRESHOLD_HOP2) &
+		PROT_BIT(mmSTLB_SET_THRESHOLD_HOP3) &
 		PROT_BIT(mmSTLB_SET_THRESHOLD_HOP4));
 	regmap_writel(mask, PRIV_PROT_REG(prot_base));
 
